terraform/modules/utils/scripts/EhrHardDeletion.py

```diff
@@ -1,11 +1,59 @@
 import json
 import boto3
 import os
+import time
+from boto3.dynamodb.conditions import Key
 
 def lambda_handler(event, context):
-    print("This is the ehr-hard-deletion-lambda!")
+    dynamodbTable, inboundConversationId = parseEvent(event)
+    deleteEhrFromS3(inboundConversationId)
+    verifyDBTableRecordsDeleted(dynamodbTable, inboundConversationId)
 
-    return {
-        'statusCode': 200,
-        'body': json.dumps('Completed!'),
-    }
+def deleteEhrFromS3(inboundConversationId):
+    s3 = boto3.resource('s3')
+    try:
+        print("Retrieving S3_REPO_BUCKET environment variable")
+        s3BucketName = os.environ["S3_REPO_BUCKET"]
+        print("S3_REPO_BUCKET=" + s3BucketName)
+        print("Retrieving S3 Bucket")
+        repoBucket = s3.Bucket(s3BucketName)
+    except KeyError as error:
+        print("Failed to get S3_REPO_BUCKET environment variable: " + error)
+    except Exception as error:
+        print("Failed to find the S3 Bucket: " + error)
+
+    if any(True for _ in repoBucket.objects.filter(Prefix=inboundConversationId + "/")): # Checks to see objects exist in the S3 Bucket for the given inboundConversationId
+        print('EHR found in the S3 Bucket')
+        try:
+            print("Attempting to delete EHR in the S3 Bucket")
+            repoBucket.objects.filter(Prefix=inboundConversationId + "/").delete()
+            if all(False for _ in repoBucket.objects.filter(Prefix=inboundConversationId + "/")):
+                print("EHR has been deleted from the S3 Bucket successfully!")
+            else:
+                print("EHR deletion was unsuccessful")
+        except Exception as error:
+            print("Failed to delete EHR in the S3 Bucket: " + error)
+    else:
+        print("EHR could not be found in the S3 Bucket")
+
+def parseEvent(event):
+    print("Parsing event")
```

Comment:
Can we utilise a logger here over print?

https://docs.python.org/3/library/logging.html

Reply:
As a side note:
When logging in a lambda the print statement automatically logs to cloudwatch in AWS. The only added benefit of using a logger in a lambda is if you want to format JSON (it is still printed with the print statement just not formatted nicely) or if you want the extra info of the log level.

see here for details https://docs.aws.amazon.com/lambda/latest/dg/python-logging.html

Reply:
Interesting note @JoeSykesNHS - this went past me :)